### PR TITLE
Window Mutex: Document that we properly initialize the SRWLock

### PR DESCRIPTION
**Pull request by RalfJung**
See https://github.com/rust-lang/rust/issues/35836

### CODE REVIEW

**Comment by RalfJung** [comment]
> The system is going down for power off NOW!

https://github.com/travis-ci/travis-ci/issues/4924

File: src/libstd/sys/windows/mutex.rs [hunk 1]
```diff
@@ -58,6 +58,8 @@ pub unsafe fn raw(m: &Mutex) -> c::PSRWLOCK {
 impl Mutex {
     pub const fn new() -> Mutex {
         Mutex {
+            // This works because SRWLOCK_INIT is a NULL pointer, so we are also properly
```
Comment (retep998) [inline src/libstd/sys/windows/mutex.rs]: Technically `SRWLOCK` is a struct wrapping a pointer which isn't actually used as a pointer.

Comment (RalfJung) [reply src/libstd/sys/windows/mutex.rs]: Oh wow. I guess Windows API doesn't really care about meaningful types?

Comment (RalfJung) [reply src/libstd/sys/windows/mutex.rs]: Fixed, I hope.

**Review by retep998 — APPROVED** [review]

**Comment by retep998** [comment]
As much as I'd love to r+ this PR, I don't have such permissions, so someone else will have to actually r+ it.

**Pull request merged**